presto-hive/src/main/java/io/prestosql/plugin/hive/parquet/ParquetPageSourceFactory.java

```diff
@@ -163,13 +168,12 @@ public static ParquetPageSource createParquetPageSource(
             MessageType fileSchema = fileMetaData.getSchema();
             dataSource = buildHdfsParquetDataSource(inputStream, path, fileSize, stats);
 
-            List<org.apache.parquet.schema.Type> fields = columns.stream()
+            List<org.apache.parquet.schema.Type> parquetFields = columns.stream()
                     .filter(column -> column.getColumnType() == REGULAR)
                     .map(column -> getParquetType(column, fileSchema, useParquetColumnNames))
-                    .filter(Objects::nonNull)
```

Comment:
Mark list as `List<@Nullable org.apache.parquet.schema.Type>`.
Better yet -- use `List<Optional<org.apache.parquet.schema.Type>>` and end this with:
```
.map(Optional::ofNullable)
.collect(toImmutableList());
```